crates/cdk/src/wallet/auth/auth_wallet.rs

```diff
@@ -292,6 +292,22 @@ impl AuthWallet {
 
         let keys = self.get_keyset_keys(active_keyset_id).await?;
 
+        // Verify the signature DLEQ is valid
+        {
+            for (sig, premint) in mint_res.signatures.iter().zip(&premint_secrets.secrets) {
+                let keys = self.get_keyset_keys(sig.keyset_id).await?;
+                let key = keys.amount_key(sig.amount).ok_or(Error::AmountKey)?;
+                match sig.verify_dleq(key, premint.blinded_message.blinded_secret) {
+                    Ok(_) => (),
+                    Err(nut12::Error::MissingDleqProof) => {
+                        // TODO: Do we want to error here and require dleq proofs?
+                        tracing::warn!("Signature for bat returned without dleq proof.");
+                    }
```

Comment:
Depending on final wording of spec we should error here or allow proofs without a dleq proof.